src/components/UI/Controls/ConfirmationPrompt/index.tsx

```diff
@@ -0,0 +1,138 @@
+import { Box, Collapsible, Keyboard, Text } from 'grommet';
+import PropTypes from 'prop-types';
+import React, { useEffect, useRef } from 'react';
+
+import Button from '../Button';
+
+interface IConfirmationPromptProps
+  extends Omit<React.ComponentPropsWithRef<typeof Collapsible>, 'title'> {
+  title?: React.ReactNode;
+  confirmButton?: React.ReactNode;
+  cancelButton?: React.ReactNode;
+  onConfirm?: () => void;
+  onCancel?: () => void;
+}
+
+const ConfirmationPrompt = React.forwardRef<
+  HTMLDivElement,
+  IConfirmationPromptProps
+>(
+  (
+    {
+      children,
+      title,
+      confirmButton,
+      cancelButton,
+      onConfirm,
+      onCancel,
+      ...props
+    },
+    ref
+  ) => {
+    const wrapperRef = useRef<HTMLDivElement>(null);
+    const cancelButtonRef = useRef<HTMLButtonElement>(null);
+
+    useEffect(() => {
+      /**
+       * Focus a control after the confirmation is opened and
+       * all the microtasks have finished running.
+       *  */
+      setTimeout(() => {
+        if (!props.open) return;
+
+        if (cancelButtonRef.current) {
+          // Focus on the cancel button if it exists.
+          const cancelButtonElement = cancelButtonRef.current.querySelector<HTMLButtonElement>(
+            '.cancel-button'
+          );
+          cancelButtonElement?.focus();
+        } else if (wrapperRef.current) {
+          // Otherwise focus on the wrapper element.
+          wrapperRef.current.focus();
+        }
```

Comment:
When we open the confirmation prompt, if the cancel button exists, it will be focused. If it doesn't exist, then the wrapper element will be focused (so we could press `Esc` and close it)